### PR TITLE
Toolchain build

### PR DESCRIPTION
**Pull request by Shillaker**
Related to https://github.com/faasm/faasm/issues/314

### CODE REVIEW

File: tasks/env.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+SYSROOT_DOCKERFILE = join(PROJ_ROOT, "docker", "sysroot.dockerfile")
+
+USABLE_CPUS = int(cpu_count()) - 1
+
+TOOLCHAIN_BIN = "/usr/local/faasm/toolchain/bin"
+SYSROOT = "/usr/local/faasm/llvm-sysroot"
+
+WASM_CC = join(TOOLCHAIN_BIN, "clang")
+WASM_AR = join(TOOLCHAIN_BIN, "llvm-ar")
+WASM_NM = join(TOOLCHAIN_BIN, "llvm-nm")
```
Comment (Shillaker) [inline tasks/env.py:21]: There is some duplicated info in here that is held elsewhere, but consolidating everything will be covered in #6

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Faasm Toolchain
+# Faasm Toolchain ![Toolchain tests](https://github.com/faasm/faasm-toolchain/workflows/Build/badge.svg) ![License](https://img.shields.io/github/license/faasm/faasm-toolchain.svg)
```
Comment (csegarragonz) [inline README.md:1]: If we have dem cool badges we ought to have a cool logo as well

File: docker/sysroot.dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+ARG LLVM_VERSION
+FROM faasm/toolchain:${LLVM_VERSION}
```
[the remark this answers is not in the document]
Comment (csegarragonz) [reply docker/sysroot.dockerfile]: Afaict, `bin/here.sh#L20` tags the image with the `faasm-toolchain` version.

File: tasks/container.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    )
+
+
+@task
+def toolchain(ctx, nocache=False, push=False):
+    """
+    Build current version of the toolchain container
+    """
+    llvm_version = get_llvm_version()
+    tag_name = "{}:{}".format(TOOLCHAIN_IMAGE_NAME, llvm_version)
```
Comment (csegarragonz) [inline tasks/container.py:65]: But here we are using the `llvm_version` as tag again.

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+name: Build
+
+on:
+  push:
+    branches: [master]
+  pull_request:
```
Comment (csegarragonz) [inline .github/workflows/build.yml:6]: Do we want to set up that this only happens on non-draft PRs?

**Review by csegarragonz — CHANGES_REQUESTED** [review]

File: docker/sysroot.dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (csegarragonz) [reply docker/sysroot.dockerfile]: Is `here.sh` removed? Then this might be outdated now.

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: "Build and push sysroot container"
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          file: docker/sysroot.dockerfile
+          tags: faasm/sysroot:${{ env.TAG_VERSION }}
```
Comment (csegarragonz) [inline .github/workflows/release.yml:33]: But here we indeed tag it with our version don't we?

**Review by csegarragonz — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 5 above)
Comment (Shillaker) [reply .github/workflows/build.yml:6]: Ah yes, I guess that would be a good idea, I'll add that in my next PR once I've added in all the other submodules

File: docker/sysroot.dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (Shillaker) [reply docker/sysroot.dockerfile]: Yes, sorry I gave up on the `here.sh` script as this container will never really be run directly. 

We're tagging the `toolchain` container with LLVM version as it only changes with LLVM versions, whereas the sysroot might change many times between LLVM versions, so it has its own numbering scheme.

**Pull request merged**